src/parse.rs

```diff
@@ -100,30 +112,50 @@ impl LinkType {
     }
 }
 
+/// Markdown events that are generated in a preorder traversal of the document
+/// tree, with additional `End` events whenever all of an inner node's children
+/// have been visited.
 #[derive(Clone, Debug, PartialEq)]
 pub enum Event<'a> {
+    /// Start of a tagged element. Events that are yielded after this event
+    /// and before its corresponding `End` event are inside this element.
+    /// Start and end events are guaranteed to be balanced.
     Start(Tag<'a>),
+    /// End of a tagged element.
     End(Tag<'a>),
+    /// A text node.
     Text(CowStr<'a>),
+    /// An inline code node.
     Code(CowStr<'a>),
+    /// An HTML node.
     Html(CowStr<'a>),
-    InlineHtml(CowStr<'a>),
+    /// An inline HTML node.
+    InlineHtml(CowStr<'a>), // TODO: do we need a distinction between Html blocks/ Html/ InlineHtml?
```

Comment:
Yes please. I'm still wondering what the difference between `Html` and `InlineHtml` is and I guess most people won't read into the CommonMark spec to find out.